internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlyStoreConfigRepository.java

```diff
@@ -114,12 +100,25 @@ public Optional<StoreConfig> getStoreConfig(String storeName) {
     if (systemStoreType != null && systemStoreType.equals(VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE)) {
       veniceStoreName = VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE.extractRegularStoreName(storeName);
     }
-    StoreConfig config = storeConfigMap.get().get(veniceStoreName);
-    if (config != null) {
-      return Optional.of(config.cloneStoreConfig());
-    } else {
-      return Optional.empty();
+
+    if (availableStoreSet.get().contains(veniceStoreName)) {
+      StoreConfig config = loadedStoreConfigMap.get().get(veniceStoreName);
+      if (config == null) {
+        // lazy fetch from ZK and attach watch
+        config = accessor.getStoreConfig(veniceStoreName);
+        if (config != null) {
+          loadedStoreConfigMap.get().put(config.getStoreName(), config);
+          accessor.subscribeStoreConfigDataChangedListener(veniceStoreName, storeConfigChangedListener);
+          StoreConfig configToVerify = accessor.getStoreConfig(veniceStoreName);
+          if (!configToVerify.equals(config)) {
+            LOGGER.debug("Store config is changed during fetching. Reload the store config.");
+            config = configToVerify;
```

Comment:
If it is truly different, we need to update the map, right?

Reply:
Ooops, missed that part

Reply:
Thanks for catching it! Also added one unit test.